pkg/config/database.go

```diff
@@ -141,9 +141,29 @@ func (d *Database) Validate() error {
 		return unknownDbType(d.Type)
 	}
 
+	if d.Host == "" {
+		return errors.New("database host missing")
+	}
+
+	if !d.isUnixAddr() && d.Port == 0 {
+		return errors.New("database port missing (or host must be a Unix domain socket)")
```

Comment:
Ermm... I'd say `(and host is not...`.

Reply:
Your proposal basically rephrases the condition. My intention with this wording was to tell the user something like "alternatively, if you don't want to specify a port number, you can use a Unix domain socket instead" (without making the text excessively long).

Anyways, I hope to get rid of this specific check rather soon (#476).

Reply:
Look.. I'd actually prefer my way, but

1. this is not super-dramatic
2. you named it (#476)

Change it. Or merge it. Your choice.

Reply:
My preferred strategy is to merge #483 first and then remove this message from this PR altogether.